File: frontend/src/app/models/flight.ts
export interface Airport {
    text: string,
    value: string
}

export interface Flight {
    id: number,
    originLocationCode: string;
    destinationLocationCode: string;
    departureDate: Date,
    returnDate?: Date | null,
    passengerCount: number,
    transferCount: number,
    currencyCode: string,
    time: number
    price: number
}

export class Flight implements Flight {
    constructor(init?: FlightFormValues) {
        Object.assign(this, init);
    }
}

export class FlightFormValues {
    originLocationCode: string = '';
    destinationLocationCode: string = '';
    departureDate: Date | null = null;
    returnDate?: Date | null = null;
    passengerCount: number = 0;
    transferCount: number = 0
    currencyCode: string = '';
    price: number = 0;

    constructor(flight: Flight) {
        this.originLocationCode = flight.originLocationCode;
        this.destinationLocationCode = flight.destinationLocationCode;
        this.departureDate = flight.departureDate;
        this.returnDate = flight.returnDate;
        this.passengerCount = flight.passengerCount;
        this.transferCount = flight.transferCount;
        this.currencyCode = flight.currencyCode;
        this.price = flight.price;
    }
}

export interface FlightSearchFormValues {
    originLocationCode: string;
    destinationLocationCode: string;
    departureDate: Date | null;
    returnDate: Date | null;
    passengerCount: number;
    currencyCode: string;
}

File: src/main/java/com/sky/getyourway/entities/Flight.java
package com.sky.getyourway.entities;


import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

import javax.persistence.*;
import java.time.LocalDate;

@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
@Inheritance
@Table(name = "flight", schema = "getyourway")
public class Flight extends Transport {

    private String flightNumber;

    @Column(name = "origin_location_code", nullable = false)
    private String originLocationCode;

    @Column(name = "destination_location_code", nullable = false)
    private String destinationLocationCode;

    @Column(name = "departure_date", nullable = false)
    private LocalDate departureDate;

    @Column(name = "return_date")
    private LocalDate returnDate;

    @Column(name = "passenger_count", nullable = false)
    private Integer passengerCount;

    @Column(name = "transfer_count", nullable = false)
    private Integer transferCount;

    @Column(name = "currency_code", nullable = false)
    private String currencyCode;

    @ToString.Exclude
    @ManyToOne
    @JoinColumn(name = "journey_id", nullable = false)
    private Journey journey;
}
